File: problems/11-heap/001-kth-largest-element/java/Solution.java
import java.util.PriorityQueue;

public class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();

        for (int value : nums) {
            minHeap.offer(value);

            if (minHeap.size() > k) {
                minHeap.poll(); // Remove the smallest of the current top k.
            }
        }

        return minHeap.peek();
    }
}

File: problems/11-heap/001-kth-largest-element/python/solution.py
import heapq


class Solution:
    def findKthLargest(self, nums: list[int], k: int) -> int:
        min_heap = []

        for value in nums:
            heapq.heappush(min_heap, value)
            if len(min_heap) > k:
                heapq.heappop(min_heap)

        return min_heap[0]
